python/tests/unit/test_config.py

```
@@ -130,3 +134,127 @@ def test_load_config_file_not_found(self):
+
+    def test_truenas_config_missing_url(self):
+        """Missing TrueNAS URL raises ConfigurationError."""
+        config = Config.__new__(Config)
+        config.data = {"truenas": {"api_key": "secret"}}
+
+        with pytest.raises(ConfigurationError, match="TrueNAS URL is required"):
+            config.truenas_config()
+
+    def test_normalize_cluster_config_rejects_non_mapping(self):
+        """Non-mapping kubernetes section raises ConfigurationError."""
+        with pytest.raises(ConfigurationError, match="kubernetes.*mapping"):
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)